src/Package/Impl/Repl/Commands/ReplCommandController.cs

```diff
@@ -152,10 +152,12 @@ public ReplCommandController(ITextView textView, ITextBuffer textBuffer)
             }
         }
 
-        private void HandleCancel(RCompletionController controller) {
-            Workflow.Operations.CancelAsync().DoNotWait();
-            // Post interrupt command which knows if it can interrupt R or not
-            VsAppShell.Current.PostCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdInterruptR);
+        private void HandleCancel(RCompletionController controller, ITextView textView) {
+            if (!controller.HasActiveCompletionSession && !controller.HasActiveSignatureSession(textView)) {
```

Comment:
I don't understand this. `InterruptRCommand` command calls `_session.CancelAllAsync`, which is also called from `Workflow.Operations.CancelAsync()`. Why do we call it twice?